src/tooling/grain_doc.md

```diff
@@ -0,0 +1,109 @@
+---
+title: grain doc
+---
+
+Grain ships with a documentation generator built directly into the compiler!
+
+You may be used to similar tools, such as `JSDoc`, that allow you to add a comment above code which describes the input and output types, along with other documentation. Generally, you'll run an external tool against your source code to generate your documentation.
+
+In Grain, we have provided this as another command via the CLI, and it inherits the full-power of the compiler. You won't need to specify any types in your doc comments, as the compiler will infer them and inserts them into the output. 🎉
+
+Here's a quick example of adding a doc block to your grain code:
+```gr
+/**
+ * An alias for normal syntactic array access, i.e. `array[n]`.
+ *
+ * Retrieves the element from the array at the specified index.
+ * A negative index is treated as an offset from the end of the array.
+ *
+ * @param index: The index to access
+ * @param array: The array to access
+ * @returns The element from the array
+ *
+ * @since v0.1.0
+ * @history v0.2.0: Argument order changed to data-last
+ */
+export let get = (index, array) => {
+  array[index]
+}
+```
+
+## Using GrainDoc
+
+### How to use GrainDoc
+Graindoc is rather simple to use you start by adding the graindoc comments marked by `/** */` note the generator only generates documentation for exports of a module to your code and then you run `grain doc <file> -o <file>` to generate the documentation. 
+
+### The Benefits Of GrainDoc
+The benefits of tools like graindoc vs hand generated documentation is, Graindoc can actually use Grain’s strong type system in order to figure out the specific types of these and inject them into the output! Your type signatures will always be up-to-date with Graindoc because compilers are much better at type systems than humans.
```

Comment:
```suggestion
```
I like this, but I think we can drop it. Maybe we can mention "Your type signatures will always be up-to-date with Graindoc because compilers are much better at type systems than humans." to explain why `@param` or `@return` don't need types (below).